jwst/extract_1d/extract.py

```diff
@@ -1204,15 +1204,23 @@ def get_target_coordinates(
         targ_dec : float or None
             The declination of the target, or None
         """
-        if slit is None:
-            targ_ra = input_model.meta.target.ra
-            targ_dec = input_model.meta.target.dec
-        else:
+        if slit is not None:
+            # If we've been passed a slit object, get the RA/Dec
+            # from the slit source attributes
             targ_ra = getattr(slit, 'source_ra', None)
             targ_dec = getattr(slit, 'source_dec', None)
             if targ_ra is None or targ_dec is None:
                 targ_ra = input_model.meta.target.ra
                 targ_dec = input_model.meta.target.dec
+        elif isinstance(input_model, datamodels.SlitModel):
```

Comment:
Since SlitModel also uses `getattr`, the situation that either are `None` still exists and should have the same check as the `slit` case. Suggested code:
```
targ_ra = None
targ_dec = None
if slit is not None:
    targ_ra = getattr...
    targ_de = getattr...
elif isinstance(input_model....
   targ_ra = getattr...
   targ_dec = getattr...
if targ_ra is None or targ_dec is None:
    targ_ra = input_model...
    targ_dec = input_model...
```

Reply:
Fixed in latest commit.